src/Microsoft.Health.Fhir.Shared.Api/Modules/FhirModule.cs

```diff
@@ -37,13 +38,7 @@ namespace Microsoft.Health.Fhir.Api.Modules
     /// </summary>
     public class FhirModule : IStartupModule
     {
-        private readonly FeatureConfiguration _featureConfiguration;
-
-        public FhirModule(FhirServerConfiguration fhirServerConfiguration)
-        {
-            EnsureArg.IsNotNull(fhirServerConfiguration, nameof(fhirServerConfiguration));
-            _featureConfiguration = fhirServerConfiguration.Features;
-        }
+        private static readonly Regex MessageChecker = new Regex("Type checking the data: Literal '(.*)' cannot be parsed as a (.*). \\(at (.*)\\)", RegexOptions.Compiled);
```

Comment:
nit: Capture groups might look cleaner
```suggestion
        private static readonly Regex MessageChecker = new Regex("Type checking the data: Literal '(?<value>.*)' cannot be parsed as a (?<type>.*). \\(at (?<location>.*)\\)", RegexOptions.Compiled);
```